doozer/doozerlib/cli/images_streams.py

```diff
@@ -886,6 +887,31 @@ def images_streams_prs(runtime, github_access_token, bug, interstitial, ignore_c
     checked_upstream_images = set()  # A PR will not be opened unless the upstream image exists; keep track of ones we have checked.
     errors_raised = False  # If failures are found when opening a PR, won't break the loop but will return an exit code to signal this event
 
+    def update_gomod(desired_ci_build_root_image, repo_dir, image_config):
+        # See if go.mod needs a change
+        golang = ''
+        try:
+            image_info = json.loads(exectools.cmd_assert(f'oc image info -o json {desired_ci_build_root_image}')[0])
+        except (KeyError, ChildProcessError):
+            logger.exception(f'Did not manage to look up {desired_ci_build_root_image}. Continuing')
+            return
+
+        for env in image_info.get('config', {}).get('config', {}).get('Env', []):
+            key, val = env.split(sep='=', maxsplit=1)
+            if key == 'VERSION' and re.match(r'^[0-9]+\.[0-9]+$', val):
+                golang = val
+
+        if not golang:
+            return
+
+        gomods = [Path(repo_dir, f'{p}/go.mod') for p in image_config.cachito.packages.gomod]
+        if not gomods:
+            gomods = [Path(repo_dir, 'go.mod')]
+        for gomod in gomods:
+            if gomod.is_file():
+                exectools.cmd_assert(f'go mod edit -go={golang} {gomod}')
```

Comment:
so this would mean a go binary running on buildvm.. 
Can we run this in hackspace / or something less risky but still tests this ?

Reply:
I see it on buildvm so good with it